Python/thread_nt.h

```diff
@@ -207,29 +207,59 @@ PyThread_start_new_thread(void (*func)(void *), void *arg)
         /* I've seen errno == EAGAIN here, which means "there are
          * too many threads".
          */
-        int e = errno;
-        threadID = (unsigned)-1;
         HeapFree(GetProcessHeap(), 0, obj);
+        return -1;
     }
-    else {
-        CloseHandle(hThread);
+    *ident = threadID;
+    *handle = (Py_uintptr_t) hThread;
+    return 0;
+}
+
+unsigned long
+PyThread_start_new_thread(void (*func)(void *), void *arg) {
+    Py_uintptr_t handle;
+    unsigned long long ident;
+    if (PyThread_start_joinable_thread(func, arg, &ident, &handle)) {
+        return PYTHREAD_INVALID_THREAD_ID;
     }
-    return threadID;
+    CloseHandle((HANDLE) handle);
+    return ident;
```

Comment:
The compiler warnings here echo my thoughts. 😄

It looks like Windows gave us an `unsigned int`.  We cast it to `unsigned long long` to "return" it from `PyThread_start_joinable_thread()`, but we still know it will fit in `unsigned long` here.

So either:

```suggestion
    return Py_SAFE_DOWNCAST(ident, unsigned long long, unsigned long);
```

or just:

```suggestion
    return (unsigned long)ident;
```

Reply:
...or turn `PyThread_start_joinable_thread()` into local function with a signature that doesn't up-cast, kind of like we do in thread_pthread.h.  For that matter, we could also preserve the original handle type as long as possible (again, like we do in thread_pthread.h).

```C
static int
do_start_joinable_thread(void (*func)(void *), void *arg,
                                        unsigned int* ident, HANDLE *handle)
{
    ...
}

int
PyThread_start_joinable_thread(void (*func)(void *), void *arg,
                               unsigned long long* ident, Py_uintptr_t* handle)
{
    Handle hThread;
    if (do_start_joinable_thread(func, arg, ident, &hThread)) {
        return -1;
    }
    *handle = (Py_uintptr_t) hThread;
    assert(hThread == (HANDLE) *handle);
    return 0;
}

unsigned long
PyThread_start_new_thread(void (*func)(void *), void *arg) {
    Handle hThread;
    unsigned long long ident;
    if (do_start_joinable_thread(func, arg, &ident, &hThread)) {
        return PYTHREAD_INVALID_THREAD_ID;
    }
    CloseHandle(hThread);
    return ident;
}
```